### ISSUE TITLE
GitLab SAST Report: ERROR [dojo:599] Error in parser: 'confidence'

### ISSUE THREAD

**Issue by mhalt** [DefectDojo/django-DefectDojo#3650]
**Bug description**
GitLab SAST reports generated from .NET, PHP, JS, etc. will fail to parse because they do not contain a scanner *confidence* key, which is required by the parser. I have included a sample report and the associated stack trace. 
https://github.com/DefectDojo/django-DefectDojo/blob/b1ef546a6a2aaf029fab594f9b68db1be83947ba/dojo/tools/gitlab_sast/parser.py#L93 

```
[18/Jan/2021 18:33:20] ERROR [dojo:598] 'confidence'
Traceback (most recent call last):
  File "./dojo/engagement/views.py", line 591, in import_scan_results
    parser = import_parser_factory(file, t, active, verified)
  File "./dojo/tools/factory.py", line 278, in import_parser_factory
    parser = GitlabSastReportParser(file, test)
  File "./dojo/tools/gitlab_sast/parser.py", line 14, in __init__
    self.items = [data for data in self.get_items(tree, test)]
  File "./dojo/tools/gitlab_sast/parser.py", line 32, in get_items
    item = get_item(node, test)
  File "./dojo/tools/gitlab_sast/parser.py", line 93, in get_item
    scanner_confidence = Finding.get_number_severity(vuln['confidence'])
KeyError: 'confidence'
[18/Jan/2021 18:33:20] ERROR [dojo:599] Error in parser: 'confidence'
```

**Steps to reproduce**
1. Run a PHP (for example because it does not have 'confidence' analyzer data) SAST scan in GitLab and get the report. 
2. Try to import this scan report into Defect Dojo. 
3. See parsing error.
4. `docker logs django-defectdojo_uwsgi_1 2>&1 | less +G`

**Expected behavior**
The scan results should be imported. 

**Deployment method**
- [X ] Docker

**Environment information**
 - DefectDojo Commit Message: [use `git show -s --format="[%ci] %h: %s [%d]"`]
There is only one version of the GitLab SAST parser: 152b1aa25b544f03f3b426f23e4812f49964d2df

**Sample scan files** (optional)
If applicable, add sample scan files to help reproduce your problem.
[gitlab-sast-report.zip](https://github.com/DefectDojo/django-DefectDojo/files/5832131/gitlab-sast-report.zip)

**Additional context** (optional)
The problem is that not all of the SAST analyzers that GitLab uses leverage the scanner *confidence* key. See below: https://docs.gitlab.com/ee/user/application_security/sast/analyzers.html#analyzers-data
![image](https://user-images.githubusercontent.com/18124608/104957169-a3552000-599b-11eb-8788-4c242fd6620a.png)

**Comment by damiencarol** [comment]
@mhalt thanks for reporting. Is it possible to have the report file that produce this error?

**Comment by StefanFl** [comment]
Pull request #3567 addresses this issue already. It has been merged to dev a few days ago. I imported the sample scan file, provided with this issue, in the dev version and it worked.

**Comment by mhalt** [comment]
@StefanFl That's great to hear! Do you know when a release is planned for this fix?

**Comment by damiencarol** [comment]
@mhalt there are usually few releases by month. The next release is planned for next week.

**Comment by damiencarol** [comment]
@mhalt we just released `1.12.0`. I'm closing this issue
